tests/e2e-evm/test/connect.test.ts

```diff
@@ -0,0 +1,50 @@
+import hre from "hardhat";
+import { expect } from "chai";
+import { whaleAddress, userAddress } from "./addresses";
+
+describe("Viem Setup", function () {
+  it("getChainID() matches the configured network", async function () {
+    const publicClient = await hre.viem.getPublicClient();
+
+    const expectedChainId = (() => {
+      switch (hre.network.name) {
+        case "hardhat":
+          return 31337;
+        case "kvtool":
+          return 8888;
```

Comment:
similar to above, I think this will be a candidate for a config object if other config is needed for either case

Reply:
Will be added to future work